go/vt/vttablet/tabletmanager/vreplication/utils.go

```diff
@@ -123,6 +125,9 @@ func isUnrecoverableError(err error) bool {
 	if err == nil {
 		return false
 	}
+	if vterrors.Code(err) == vtrpc.Code_INTERNAL {
+		return true
+	}
```

Comment:
This has potentially far reaching effects: https://github.com/vitessio/vitess/blob/8575b170183bce389103d451965dc12070f4e36d/proto/vtrpc.proto#L158-L161

I typically use that when there's something very unexpected, which may be e.g. some random bit flip or unknown edge case that would be fine on retry. Please see the error code discussion above as we should use a code which indicates that you cannot simply retry w/o fixing the underlying issue (FAILED_PRECONDITION seems like the best one).

This will affect all vreplication usage and will almost certainly lead to some unintended changes.

Reply:
Not debating which name is more appropriate, but I did a quick grep and we are using `Code_internal` today in both vstreamer and vreplication for conditions that are non-recoverable. `FAILED_PRECONDITION` is not used anywhere as yet.

Reply:
This makes sense. I'll change to `FAILED_PRECONDITION`.